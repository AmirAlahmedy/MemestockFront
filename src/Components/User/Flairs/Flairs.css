.labels{
    display: flex;
    justify-content: center;
    align-items: center;
    align-content: center;
}

.list{
    display: flex;
    justify-content: center;
    align-items: center;
    align-content: center;
    padding-top: 10px;
}