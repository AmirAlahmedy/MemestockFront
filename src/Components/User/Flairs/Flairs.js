import React, { Component } from 'react';
import './Flairs.css';
import axios from '../../../axios-orders';





class Flairs extends Component {
        state = {
                // flairs: [
                //         'ali'
                // ],
                // correspondingSubreddits: [
                //         'test'
                // ]
                response: []
        }
        header = {
                'auth': localStorage.getItem('token')
            }
        componentDidMount(){
                console.log('token for flairs',this.header.auth);
                axios.get('user/Flairs', {headers: this.header})
                    .then( res => {
                            console.log(res);
                        this.setState({
                                response: res.data
                        })
                            
                    })
                    .catch( error => {

                if(error.status === 404){

                        this.setState({
                                // flairs: [],
                                // correspondingSubreddits: []
                                response: []
                        })
                        
                }
                })
            }

        /**
         * For generating flairs
         * @function getFlairs
         */
         getFlairs() {
        if (this.state.response === 0) return;
        return this.state.response.map(el => <Flair
                flair={el.flair}
                sub={el.srName}
        />)
        }

        render() {
                console.log(this.state);
             if( this.state.response.length > 0 ){

                     return(
                         <div>{this.getFlairs()}</div>
                        );
                }else{
                        return(
                                <em style={{color: 'red'}}>You have no flairs</em>
                        );
                }
        }
} 
class Flair extends Component{

        render(){
                return(
                        <div>Flair: {this.props.flair} Subreddit: {this.props.sub}</div>
                        );
        }
}
export default Flairs;