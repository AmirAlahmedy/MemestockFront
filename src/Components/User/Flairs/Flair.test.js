import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import axios from 'axios';
import Flairs from './Flairs';
import { create } from "react-test-renderer";

configure({adapter: new Adapter()});
//jest.mock("axios");

describe('<Flairs/>', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Flairs></Flairs>);
    });

    test("it matches the snapshot", () => {
        const component = create(<Flairs />);
        expect(component.toJSON()).toMatchSnapshot();
      });

    it("shows a list of flairs", async () => {
        const component = create(<Flairs />);
        const instance = component.getInstance();
        await instance.componentDidMount();
        console.log(instance.state) 
    });

    // it("shows a list of flairs (mock)", async () => {
    //     const response = {
    //       flairs:  [
    //         {
    //             "_id": "5cc5e38d58af9d0e34be6663",
    //             "username": "mostafa_hazem",
    //             "srName": "Education",
    //             "flair": "el bob",
    //             "__v": 0
    //         },
    //         {
    //             "_id": "5cc5ed2bf14172494416af83",
    //             "username": "mostafa_hazem",
    //             "srName": "Technology",
    //             "flair": "el bob2",
    //             "__v": 0
    //         }
    //     ]

    //     };
    //     axios.get.mockResolvedValue(response);
    //     const component = create(<Flairs />);
    //     const instance = component.getInstance();
    //     await instance.componentDidMount();
    //     console.log(instance.state); // << HERE IS THE SNITCH!
    //   });

    
})

