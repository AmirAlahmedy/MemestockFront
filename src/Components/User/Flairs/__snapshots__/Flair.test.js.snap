// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flairs/> it matches the snapshot 1`] = `
<em
  style={
    Object {
      "color": "red",
    }
  }
>
  You have no flairs
</em>
`;
